downloader/src/main/kotlin/VersionControlSystem.kt

```diff
@@ -50,7 +51,7 @@ abstract class VersionControlSystem {
         /**
          * Return the applicable VCS for the given [vcsType], or null if none is applicable.
          */
-        fun forType(vcsType: String) = ALL.find { it.isApplicableType(vcsType) }
+        fun forType(vcsType: VcsType) = ALL.find { it.isApplicableType(vcsType) }
```

Comment:
Using an enum for the type makes it impossible to add VCS implementations via plugin, because the enum cannot be extended. Since we now know exactly what VCS implementations are available we should not use `ServiceLoader` to initialize `ALL` anymore. Instead we could hardcode the list to reduce complexity.
For this function I would propose to remove the `isApplicableType(VcsType)` function completely and instead use a `when` expression for the mapping of `VcsType` to implementation, so probably we don't need `ALL` anymore.